### DISCUSSION TITLE
How to get app credentials?

### DISCUSSION THREAD

**Discussion by nagyrobi** [astrandb/miele#209] [Q&A]
### The problem

How to get _app credentials_? At https://www.miele.com/developer site there's no such option, under the [Get Involved](https://www.miele.com/developer/getinvolved.html) link there's this message:

```
Registration for client credentials is currently not available.
We are working on the issue.
Please try again later.
```

Later when? Or am I looking in the wrong place?

### Version of miele integration having the issue?

miele 0.1.1

### Version of Home Assistant Core having the issue?

v2022.12.5

**Comment by astrandb** [comment]
You are looking in the right place! The registration service seems do be down since a couple of days.

**Comment by jvandijk** [comment]
I just sent them an email as stated on that page now and I actually received credentials from them.

**Comment by mosart** [comment]
How many days did it took to wait? ( I am waiting for two days now. I did get a confirmation email, that said: "If you would like to request client credentials, please use our online form at [developer.miele.com](https://www.miele.com/f/com/en/register_api.aspx)" ) 

The web form still does not seem to work. (That is, I don't get a "request has been submitted" after pressing the submit button.

web page: https://www.miele.com/developer/getinvolved.html 
Mentions the webform is down, and to use email instead.

webform: https://www.miele.com/f/com/en/register_api.aspx

email with pre-filled entries: [developer@miele.com](mailto:developer@miele.com?subject=Request%20for%20Client%20Credentials&body=App%20name:%0AContact:)

**Comment by nagyrobi** [comment]
I wrote on the 15th and got no answer.
Mailed again yesterday, still no answer...

**Comment by astrandb** [comment]
I know that Miele is well aware of the problems with the credentials forms etc.

**Comment by TFhdKi95ae8L** [comment]
Received credentials just now.
Requested the 29th, via email.

**Comment by nagyrobi** [comment]
Still nothing for me :(

**Comment by mosart** [comment]
I got credentials today. So, I guess it took a while because of the
Christmas holidays?

Op vr 6 jan. 2023 13:02 schreef H. Árkosi Róbert ***@***.***>:

**Comment by samikentta** [comment]
I just also got my credentials now (client id & client secret) after waiting for a week or so for my email request. 

Now, I don't get how to proceed -- I mean, when I add the Miele integration though HA settings, first step guides me to an external website (api.mcs3.miele.com...) which requires login with email & password. What exactly is the email and password supposed to be used there? I have a Miele account which works for Miele in my home country (Finland / miele.fi). Am I supposed to register somewhere else as well? Any help anyone?

**Comment by astrandb** [comment]
@samikentta Use same username/password as you have in the app. Note that if you have special characters in the password it will probably not work.

**Comment by samikentta** [comment]
@astrandb Thanks, but unfortunately login does not work for me. Login to Miele app on my phone works well (tested logout/login) and I also tried to remove all special chars from my password. Is this api login something that Miele enables only when seeking for credentials?

**Comment by astrandb** [comment]
Please try your credentials at https://www.miele.com/developer/swagger-ui/ If you can authorize there it will normally work in the integration also.

**Comment by samikentta** [comment]
Thanks! Now I'm in - Swagger UI worked, which made me realize that I in fact had once added some nonsense to client id and client secret in HA - so the integration did not pass proper id&secret to Miele login page. Issue was fixed by removing the application credentials in HA. You could call it a user error.  😳

**Comment by beasthouse-au** [comment]
> Issue was fixed by removing the application credentials in HA. 

How did you do this?

**Comment by samikentta** [comment]
@beasthouse-au Goto Home Assistant->Integrations, click on three dots in top right corner and select "Application Credentials".  Check "Miele" and then press remove seleced. 
![image](https://user-images.githubusercontent.com/25582587/212494772-ad252ba6-da6a-4a48-a862-5ee0b5a81d68.png)

**Comment by beasthouse-au** [comment]
Thanks @samikentta - now I just need to figure out why the authentication fails!

**Comment by beasthouse-au** [comment]
> How to get _app credentials_? At https://www.miele.com/developer

This is working now for me
